### PR TITLE
chore: update next-major-spec with master

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR updates next-major-spec branch with latest master changes.

Except for the json-schema files, as they will be added here: https://github.com/asyncapi/spec-json-schemas/pull/450

### CODE REVIEW

**Review by smoya — DISMISSED** [review]

**Comment by jonaslagoni** [comment]
Remember to merge this as a regular merge commit (NOT Squash)

**Review by char0n — DISMISSED** [review]

File: http/README.md [hunk 1]
```diff
@@ -30,8 +30,14 @@ This object MUST NOT contain any properties. Its name is reserved for future use
 
 Field Name | Type | Description
 ---|:---:|---
+<<<<<<< HEAD
```
Comment (derberg) [inline http/README.md:33]: some merge leftovers

File: http/README.md [hunk 2]
```diff
@@ -95,4 +101,9 @@ channels:
             bindingVersion: '0.2.0'
 ```
 
-[schemaObject]: https://www.asyncapi.com/docs/specifications/latest/#schemaObject
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (derberg) [inline http/README.md:104]: merge leftovers

**Review by derberg — CHANGES_REQUESTED** [review]
some merge leftovers left in markdown

File: http/README.md (hunk 2 above)
Comment (char0n) [reply http/README.md:104]: Being dealt with in https://github.com/asyncapi/bindings/pull/220 and https://github.com/asyncapi/bindings/pull/221

File: http/README.md [hunk 3]
```diff
@@ -30,8 +30,14 @@ This object MUST NOT contain any properties. Its name is reserved for future use
...
 Field Name | Type | Description
 ---|:---:|---
+<<<<<<< HEAD
 <a name="operationBindingObjectMethod"></a>`method` | string | The HTTP method for the request. Its value MUST be one of `GET`, `POST`, `PUT`, `PATCH`, `DELETE`, `HEAD`, `OPTIONS`, `CONNECT`, and `TRACE`.
 <a name="operationBindingObjectQuery"></a>`query` | [Schema Object][schemaObject] | A Schema object containing the definitions for each query parameter. This schema MUST be of type object and have a properties key.
+=======
+<a name="operationBindingObjectType"></a>`type` | string | **REQUIRED**. Type of operation. Its value MUST be either `request` or `response`.
+<a name="operationBindingObjectMethod"></a>`method` | string | When `type` is `request`, this is the HTTP method, otherwise it MUST be ignored. Its value MUST be one of `GET`, `POST`, `PUT`, `PATCH`, `DELETE`, `HEAD`, `OPTIONS`, `CONNECT`, and `TRACE`.
+<a name="operationBindingObjectQuery"></a>`query` | [Schema Object][schemaObject] \| [Reference Object](referenceObject) | A Schema object containing the definitions for each query parameter. This schema MUST be of type `object` and have a `properties` key.
+>>>>>>> master-github-upstream
```
Comment (jonaslagoni) [inline http/README.md:40]:
```suggestion
<a name="operationBindingObjectMethod"></a>`method` | string | The HTTP method for the request. Its value MUST be one of `GET`, `POST`, `PUT`, `PATCH`, `DELETE`, `HEAD`, `OPTIONS`, `CONNECT`, and `TRACE`.
<a name="operationBindingObjectQuery"></a>`query` | [Schema Object][schemaObject] \| [Reference Object](referenceObject) | A Schema object containing the definitions for each query parameter. This schema MUST be of type `object` and have a `properties` key.
```

File: http/README.md [hunk 4]
```diff
@@ -95,4 +101,9 @@ channels:
...
 ```
 
-[schemaObject]: https://www.asyncapi.com/docs/specifications/latest/#schemaObject
\ No newline at end of file
+<<<<<<< HEAD
+[schemaObject]: https://www.asyncapi.com/docs/specifications/latest/#schemaObject
+=======
+[schemaObject]: https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#schemaObject
+[referenceObject]: https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#referenceObject
+>>>>>>> master-github-upstream
```
Comment (jonaslagoni) [inline http/README.md:109]:
```suggestion
[schemaObject]: https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#schemaObject
[referenceObject]: https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#referenceObject
```

File: kafka/README.md [hunk 5]
```diff
@@ -145,10 +145,14 @@ This object contains information about the message representation in Kafka.
...
 <a name="messageBindingObjectSchemaIdLocation"></a>`schemaIdLocation` | string | If a Schema Registry is used when performing this operation, tells where the id of schema is stored (e.g. `header` or `payload`). | OPTIONAL | MUST NOT be specified if `schemaRegistryUrl` is not specified at the Server level
 <a name="messageBindingObjectSchemaIdPayloadEncoding"></a>`schemaIdPayloadEncoding` | string | Number of bytes or vendor specific values when schema id is encoded in payload (e.g `confluent`/ `apicurio-legacy` / `apicurio-new`). | OPTIONAL | MUST NOT be specified if `schemaRegistryUrl` is not specified at the Server level
 <a name="messageBindingObjectSchemaLookupStrategy"></a>`schemaLookupStrategy` | string | Freeform string for any naming strategy class to use. Clients should default to the vendor default if not supplied. | OPTIONAL | MUST NOT be specified if `schemaRegistryUrl` is not specified at the Server level
+>>>>>>> master
```
Comment (jonaslagoni) [inline kafka/README.md:155]:
```suggestion
<a name="messageBindingObjectKey"></a>`key` | [Schema Object][schemaObject] \| [AVRO Schema Object](https://avro.apache.org/docs/current/spec.html) | The message key. **NOTE**: You can also use the [reference object](https://asyncapi.io/docs/specifications/v2.4.0#referenceObject) way.
<a name="messageBindingObjectSchemaIdLocation"></a>`schemaIdLocation` | string | If a Schema Registry is used when performing this operation, tells where the id of schema is stored (e.g. `header` or `payload`). | OPTIONAL | MUST NOT be specified if `schemaRegistryUrl` is not specified at the Server level
<a name="messageBindingObjectSchemaIdPayloadEncoding"></a>`schemaIdPayloadEncoding` | string | Number of bytes or vendor specific values when schema id is encoded in payload (e.g `confluent`/ `apicurio-legacy` / `apicurio-new`). | OPTIONAL | MUST NOT be specified if `schemaRegistryUrl` is not specified at the Server level
<a name="messageBindingObjectSchemaLookupStrategy"></a>`schemaLookupStrategy` | string | Freeform string for any naming strategy class to use. Clients should default to the vendor default if not supplied. | OPTIONAL | MUST NOT be specified if `schemaRegistryUrl` is not specified at the Server level
```

**Comment by jonaslagoni** [comment]
Should be all fixed now @derberg

**Review by derberg — APPROVED** [review]

**Pull request merged**